### PR TITLE
Add ability to retry intermittent tasks

### PR DESCRIPTION
**Pull request by gregarndt**

### CODE REVIEW

**Comment by gregarndt** [comment]
Missing right now is the schema, I'll add that soon.

File: test/integration/retry_intermittent_task_test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        command: ['/bin/bash', '-c', 'exit 36'],
+        maxRunTime: 1 * 60,
+        onExitStatus: {
+          retry: [36]
+        }
+      }
+    });
+
+    assert(result.run.state === 'pending', 'new pending run should be created');
+    assert(result.run.reasonCreated === 'task-retry', 'run should have been created becaus eof task retry');
```
Comment (walac) [inline test/integration/retry_intermittent_task_test.js]: nit: "because of"

File: test/integration/retry_intermittent_task_test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        maxRunTime: 1 * 60,
+        onExitStatus: {
+          retry: [36]
+        }
+      }
+    });
+
+    assert(result.run.state === 'completed', 'task should not fail or be retried');
+    assert(result.run.reasonResolved === 'completed', 'task should not or be retried');
+    assert(result.status.runs.length === 1);
```
Comment (walac) [inline test/integration/retry_intermittent_task_test.js]: replace `assert` by `assert.equal` for better error reporting.

**Comment by walac** [comment]
Some tests failing...

**Comment by gregarndt** [comment]
Thanks for the comments @walac .  I've fixed up those assertions and added a schema.  Any tests that fail now should hopefully just be our normal run of the mill intermittent failures and we can land this.

**Comment by gregarndt** [comment]
Thank you sir!

**Pull request merged**